modules/core/src/test/java/org/apache/ignite/p2p/P2PCustomSqlFunctionsConfigurationTest.java

```diff
@@ -0,0 +1,351 @@
+        }
+    }
+
+    /** */
+    private ClassLoader clsLoader;
+
+    /** */
+    private CacheConfiguration<?, ?> staticCacheCfg;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setPeerClassLoadingEnabled(false);
+
+        if (clsLoader != null)
+            cfg.setClassLoader(clsLoader);
+
+        if (staticCacheCfg != null)
+            cfg.setCacheConfiguration(staticCacheCfg);
+
+        return cfg;
+    }
+
+    /**
+     * Test verifies that if a class is partially available to the servers (some servers have it on classpath,
+     * others don't) a request to start a cache with that class in configuration fails with proper error,
+     * no nodes fail or hang.
+     *
+     * @throws Exception If test fails to set up necessary environment.
+     */
+    @Test
+    @Ignore("https://ggsystems.atlassian.net/browse/GG-36341")
+    public void testClientStartsDynamicCacheWithPartiallyUnavailableClass() throws Exception {
+        clsLoader = CONFIGURATION_CLASS_LOADER;
+
+        //classloader with a specific class is available to one server...
+        startGrid(0);
+
+        IgniteEx cl0 = startClientGrid(1);
+
+        clsLoader = null;
+
+        //... but not to the other
+        startGrid(2);
+
+        try {
+            cl0.getOrCreateCache(new CacheConfiguration<>(DYNAMIC_CACHE_NAME)
+                .setSqlFunctionClasses(CONFIGURATION_CLASS_LOADER.loadClass(UNAVAILABLE_TO_SERVER_SQL_FUNCTIONS_CLASS_NAME)));
+        }
+        catch (Exception e) {
+            String errorMsg = e.getMessage();
+
+            assertTrue("Exception during cache creation is expected to be about a class not found "
+                    + "during creation process but was: '" + errorMsg + "'",
+                errorMsg.contains("ClassNotFoundException"));
+
+            assertTrue("Exception during cache creation is expected to be about a particular class: "
+                    + UNAVAILABLE_TO_SERVER_SQL_FUNCTIONS_CLASS_NAME
+                    + ", but was: '" + errorMsg + "'",
+                errorMsg.contains(UNAVAILABLE_TO_SERVER_SQL_FUNCTIONS_CLASS_NAME));
+        }
+
+        checkTopology(3);
+    }
+
+    /**
+     * Test verifies that dynamic cache request issued by a client node with a class unavailable on server side
+     * fails with a proper exception.
+     * However, no client nor server nodes fail as a result.
+     *
+     * @throws Exception If test fails to set up necessary environment.
+     */
+    @Test
+    public void testClientStartsDynamicCacheWithUnavailableClass() throws Exception {
```

Comment:
Let's add the following test scenario:
 - start one server and one client node
 - create a new dynamic cache (let's assume that the required class is available on both nodes)
 - start a new server/client node that does not contain the required class in its classpath

Reply:
Done, new scenario added and works as expected (new node fails to join the cluster).